src/cljs/harja/views/raportit.cljs

```diff
@@ -839,9 +873,10 @@
          " sekunnin kuluttua."]]))))
 
 (defn raporttivalinnat-ja-raportti []
-  (let [r @raportit/suoritettu-raportti]
+  (let [r @raportit/suoritettu-raportti
+        ensimmainen-urakka-viimeksi (atom nil)]
     [:span
-     [raporttivalinnat]
+     [raporttivalinnat ensimmainen-urakka-viimeksi]
```

Comment:
Mikä tämän "ensimmäiset tulevat viimeiseksi" tarina on?

Reply:
Joo, lore kuuluu näin...

Tämä ratkaisu voi näyttää omatekoselta, koska se on. Täällä näkymässä ei  ole tuckin hienouksia,  enkä löytänyt pikasella katsomisella navigaatiosta mitään indikaattoria siihen haetaanko urakoita valittuun hallintayksikköön, joten tämän hyrrän teko vaati  usean  eri  liikkujan. Jos on parempia ehdotuksia, otan  kyllä  vastaan.

"**ensimmainen-urakka-viimeksi**" * on atomi, joka sisältää valitun hallintayksikön ensimmäisen urakan:

![image](https://github.com/user-attachments/assets/64f33f8d-0305-44f2-886b-91ce035a46e2)

Nyt kun hallintayksikköä vaihdetaan, voimme tunnistaa urakoiden lataamisen jos '"ensimmäinen urakka" on sama kuin edellisessä hallintayksikössä. Tämä tarkoittaa että listaa päivitetään

Reply:
Miksi tämän lisäsin,  koska hallintayksikön muutoksella voi olla joskus tuotannossa useankin sekunnin viive, että käyttäjä  ehtii avata tuon urakka alasvedon vaikka  siinä on vielä edellisen hallintayksikön  urakat, sitten ne muuttuu siinä lennosta, mikä ei ole hyvä